doc/source/sphinxext/postprocess_api.py

```diff
@@ -0,0 +1,56 @@
+"""
+Sphinx plugin to run example scripts and create a gallery page.
+
+Lightly modified from the seaborn project (Michael Waskom).
+Originally, lightly modified from the mpld3 project.
+
+"""
+import os.path as op
+import glob
+from sphinx.application import Sphinx
+
+
+def on_build_finished(app: Sphinx, exception: Exception) -> None:
+    html_dir = op.abspath(app.builder.outdir)
+    api_dir = op.join(html_dir, 'api')
+
+    # Check if the index has Jekyll template marks
+    has_jekyll_marks = False
+    index_html = op.join(html_dir, 'index.html')
+    with open(index_html, 'rt') as f:
+        lines = f.readlines()
+        if sum(line == '---\n' for line in lines) == 2:
```

Comment:
We might also use `---` later in the document as a horizontal ruler (I guess that's valid Markdown, right?). So, I'd check whether the sum is at least two, and when looking for `mark_end`, note that the `list.index()` method has an argument to specify the start index to look for, and you can use `mark_start+1` there to find the second mark.

Reply:
Actually, it's easier than that: the templated `index.html` *starts* with a `---\n` line, while nontemplated HTML starts with something else (an XML tag or empty line). Changing it